snapcraft/internal/repo.py

```diff
@@ -299,6 +299,9 @@ def _get(self, apt_cache, package_names):
                 logger.debug(
                     'Marking {!r} (and its dependencies) to be fetched'.format(
                         name))
+                version = None
+                name_arch, version = _get_pkg_name_parts(name)
+                _set_candidate(apt_cache[name_arch], version)
```

Comment:
Assuming I understand this correctly, this will silently ignore the case when an invalid or non-existing version is used, and use the most recent version instead. We should probably raise an exception instead so we don't do something unexpected. What do you think? If you agree, please add a test for it as well.